components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt

```diff
@@ -413,15 +413,21 @@ internal class GeckoPromptDelegate(private val geckoEngineSession: GeckoEngineSe
 
     private fun Uri.toFileUri(context: Context): Uri {
         val uri = this
-        val temporalFile = java.io.File(context.cacheDir, uri.getFileName(context))
+        val cacheUploadDirectory = java.io.File(context.cacheDir, "/uploads")
+
+        if (!cacheUploadDirectory.exists()) {
+            cacheUploadDirectory.mkdir()
+        }
+
+        val temporalFile = java.io.File(cacheUploadDirectory, uri.getFileName(context))
         try {
-            val inStream = context.contentResolver.openInputStream(uri) as FileInputStream
-            val outStream = FileOutputStream(temporalFile)
-            val inChannel = inStream.channel
-            val outChannel = outStream.channel
-            inChannel.transferTo(0, inChannel.size(), outChannel)
-            inStream.close()
-            outStream.close()
+            (context.contentResolver.openInputStream(uri) as FileInputStream).use { inStream ->
+                FileOutputStream(temporalFile).use { outStream ->
+                    val inChannel = inStream.channel
+                    val outChannel = outStream.channel
+                    inChannel.transferTo(0, inChannel.size(), outChannel)
```

Comment:
nit: Kotlin also has [copyTo](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/java.io.-input-stream/copy-to.html) which you can use directly on the streams without having to deal with channels.

Reply:
Awesome 👍